src/rp2_common.cmake

```diff
@@ -17,10 +17,15 @@ function(pico_add_dis_output TARGET)
             )
 endfunction()
 
+function(pico_add_thumb_enforcement TARGET)
+    add_custom_command(TARGET ${TARGET} POST_BUILD COMMAND ${CMAKE_READELF} -A ${TARGET}${CMAKE_EXECUTABLE_SUFFIX} | grep Tag_ARM_ISA_use && { echo error, the cross gcc and newlib must be built with thumb multilib support \; exit 1 \; } || exit 0)
```

Comment:
Does there need to be an additional check somewhere that readelf is installed? Or does CMake do that "automatically"?

Reply:
readelf and ld are both part of the binutils package, and you must have binutils installed to link anything, and pico-sdk needs both host and target linkers, since its compiling some host utils